backend/openrouter.py

```
@@ -48,15 +105,32 @@ async def query_model(
+            error_type='timeout',
+            message=f'Request timed out after {timeout}s.',
+            model=model
+        )
+    except httpx.HTTPStatusError as e:
+        return ModelQueryError(
+            error_type='unknown',
+            message=f'HTTP error: {e}',
+            status_code=e.response.status_code if e.response else None,
+            model=model
+        )
     except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)